x/leverage/keeper/limits.go

```diff
@@ -9,11 +9,7 @@ import (
 // maxWithdraw calculates the maximum amount of uTokens an account can currently withdraw.
 // input denom should be a base token. Uses either real or historic prices.
 func (k *Keeper) maxWithdraw(ctx sdk.Context, addr sdk.AccAddress, denom string, historic bool) (sdk.Coin, error) {
-	if types.HasUTokenPrefix(denom) {
-		return sdk.Coin{}, types.ErrUToken
-	}
 	uDenom := types.ToUTokenDenom(denom)
-
```

Comment:
We should keep this - it will give a more informative error message than if we let the function proceed.

Reply:
Note: this check also applies to the maxWithdraw query

Reply:
This check is moved to `MaxWithdraw`. The reasons are:
* combine it with the Token registry check
* have it before token registry check to get a better and more logical message.

Reply:
In that case we should also add this uToken check in `QueryMaxWithdraw` in `grpc_query.go`, which used to benefit from the error check common to both the Msg and the Query

Reply:
OK. I've also added more test and a helper function `QueryMaxWithdraw.Validate()`